extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -827,21 +919,33 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
      * @param value the geometry
      */
     fun geometry(value: Geometry) = apply {
-      val propertyValue = PropertyValue("data", value.toValue())
+      rawGeoJson = value
+      val propertyValue = PropertyValue("data", "null")
       properties[propertyValue.propertyName] = propertyValue
     }
+
     /**
      * Build the GeoJsonSource.
      *
      * @return the GeoJsonSource
      */
-    fun build() = GeoJsonSource(this)
+    fun build() = GeoJsonSource(this, rawGeoJson, onGeoJsonParsed)
   }
 
   /**
    * Static variables and methods.
    */
   companion object {
+    private val workerThread by lazy {
+      HandlerThread("STYLE_WORKER").apply {
+        priority = Thread.MAX_PRIORITY
+        start()
+      }
+    }
+    private val workerHandler by lazy {
+      Handler(workerThread.looper)
+    }
+    private val mainHandler = Handler(Looper.getMainLooper())
```

Comment:
Love that we are sharing a HandlerThread across geojsonsource instances but it seems it will run forever once used? Should we introduce management of it and call https://developer.android.com/reference/android/os/HandlerThread#quit() ?

Reply:
@tobrun I completely agree, but personally I do not have an idea how it should be managed in general. What should be a correct moment to call `quit` for a handler thread? It's not tied to `MapView` lifecycle as was mentioned [here](https://github.com/mapbox/mapbox-maps-android/pull/306#discussion_r627292315).
I think that leaving worker thread alive (which is basically doing no work) is better in that case then introducing management which might bring in some corner situations. However I'm happy to discuss, we might revisit it as tailwork.

Reply:
what about using coroutines and push the thread management issue to the language? :)

Reply:
I guess it should be done a bit later, not in this PR, WDYT?